store/tikv/gcworker/gc_worker.go

```diff
@@ -324,7 +324,7 @@ func (w *GCWorker) checkPrepare(ctx context.Context) (bool, uint64, error) {
 	if err != nil || !ok {
 		return false, 0, errors.Trace(err)
 	}
-	newSafePoint, newSafePointValue, err := w.calculateNewSafePoint(ctx, now)
+	newSafePoint, newSafePointValue, err := w.calcNewSafePoint(ctx, now)
```

Comment:
This new safe point is saved before locks are resolved. Then, if we fail to resolve locks and we don't advance the global min start ts, GC will not retry.

Reply:
How about saving it after GC is done?

Reply:
I think it ok.